src/payment/checkout/payment-form/StripeCardPayment.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react';
+import {
+  PaymentElement,
+  useStripe,
+  useElements,
+} from '@stripe/react-stripe-js';
+
+export default function StripeCardPayment() {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [message, setMessage] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (!stripe) {
+      return;
+    }
+
+    const clientSecret = new URLSearchParams(window.location.search).get(
```

Comment:
I suppose you can pass this in through the Elements component as props instead of getting it through the window object, but I wonder if it's available in one of the reducers. I will check!